lib/Backend/BailOut.h

```diff
@@ -46,7 +46,8 @@ class BailOutInfo
 #if DBG
         wasCopied = false;
 #endif
-        this->capturedValues.argObjSyms = nullptr;
+        this->capturedValues = JitAnew(bailOutFunc->m_alloc, CapturedValues);
+        this->capturedValues->refCount = 1;
```

Comment:
We should have constructor set refCount as 1 to start with instead of explicit set.

Reply:
I did it this way to enforce a contract that refcount is increased/set only when something actually takes a reference to the new CapturedValues object.

Reply:
New (constructor) should always be a reference even if temporary object.

---
In reply to: [160532659](https://github.com/Microsoft/ChakraCore/pull/4520#discussion_r160532659) [](ancestors = 160532659)

Reply:
If nothing is referring to a temp object, why should its ref count be kept at 1 (and prevent potential release)?

Reply:
local reference is also a reference. I am fine if we want to keep it this way. 

---
In reply to: [160533861](https://github.com/Microsoft/ChakraCore/pull/4520#discussion_r160533861) [](ancestors = 160533861)